pkg/image/registry/imagesignature/rest.go

```diff
@@ -0,0 +1,98 @@
+package imagesignature
+
+import (
+	"errors"
+
+	kapi "k8s.io/kubernetes/pkg/api"
+	kapierrors "k8s.io/kubernetes/pkg/api/errors"
+	"k8s.io/kubernetes/pkg/api/rest"
+	"k8s.io/kubernetes/pkg/api/unversioned"
+	"k8s.io/kubernetes/pkg/runtime"
+
+	imageapi "github.com/openshift/origin/pkg/image/api"
+	"github.com/openshift/origin/pkg/image/api/validation"
+	"github.com/openshift/origin/pkg/image/registry/image"
+)
+
+// REST implements the RESTStorage interface for ImageSignature
+type REST struct {
+	imageRegistry image.Registry
```

Comment:
> The downside is that ImageInterface will have to be extended for Update() method.

That's just a client library.  `curl` can already make these call, so it isn't increasing API surface area.